aptos-indexer-processors-sdk/testing-framework/src/mock_grpc.rs

```diff
@@ -0,0 +1,77 @@
+use anyhow::Context;
+use aptos_protos::indexer::v1::{
+    raw_data_server::{RawData, RawDataServer},
+    GetTransactionsRequest, TransactionsResponse,
+};
+use futures::Stream;
+use std::pin::Pin;
+use tokio::sync::Mutex;
+use tonic::{Request, Response, Status};
+
+const GRPC_ADDRESS: &str = "127.0.0.1:51254";
+
+#[derive(Default)]
+pub struct MockGrpcServer {
+    pub transactions_response: Mutex<Vec<TransactionsResponse>>,
+    pub chain_id: u64,
+}
+
+type ResponseStream = Pin<Box<dyn Stream<Item = Result<TransactionsResponse, Status>> + Send>>;
+
+#[tonic::async_trait]
+impl RawData for MockGrpcServer {
+    type GetTransactionsStream = ResponseStream;
+
+    async fn get_transactions(
```

Comment:
It seems like this only returns 1 transaction at a time? I think it could be helpful for the txn stream to emit more than 1 transaction, like if you want to test a sequence of transactions. Or if a processor processes txn's in parallel, txn's could get out of order, and we want to test that the DB output stays the same.

Reply:
I have a follow-up PR to handle this: https://github.com/aptos-labs/aptos-indexer-processor-sdk/pull/70

Reply:
and out of order txns, I can also follow-up to permute